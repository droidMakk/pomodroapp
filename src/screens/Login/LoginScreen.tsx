import React, { ReactNode } from "react";
import { Card } from "primereact/card";
import { Button } from "primereact/button";
import { Input } from "../../components/common";

interface LoginProps {}

interface LoginState {
	loginClicked: boolean;
	loginDisabled: boolean;
	inputs: ILoginFields;
}

interface ILoginFields {
	username: string;
	password: string;
}

export interface ILoginScreen {
	title: string;
	subTitle: string;
	state: LoginState;
	handleInput: (type: string) => (e: React.FormEvent<HTMLInputElement>) => void;
	validateInput: (inputs: ILoginFields) => boolean;
	footer: ReactNode;
}

export class LoginScreen extends React.Component<LoginProps, LoginState> implements ILoginScreen {
	title = "Login";
	subTitle = "Enter your credentials to get started!";
	state: LoginState = {
		loginClicked: false,
		loginDisabled: false,
		inputs: { username: "", password: "" }
	};

	handleInput = (type: string) => {
		console.info("TYPE", type);
		return (e: React.FormEvent<HTMLInputElement>) => {
			console.info("VALUE", e.currentTarget.value);

			let state = this.state;
			state.inputs = {
				...state.inputs,
				[type]: e.currentTarget.value
			};
		};
	};

	validateInput = (inputs: ILoginFields): boolean => {
		let isSubmittable = true,
			testRegex = /^$|\s+/;

		isSubmittable = !testRegex.test(inputs.username) && !testRegex.test(inputs.password);

		return isSubmittable;
	};

	footer = () => (
		<span>
			<Button label="Login" icon="pi pi-times" className="p-button-secondary" disabled={false} />
		</span>
	);

	handleSubmit = (hasError: boolean, e: React.FormEvent<HTMLFormElement>) => {
		let formdata = new FormData(e.currentTarget);
		console.info("FORM VALUES", formdata.get("username"), formdata.get("password"));
	};

	render() {
		return (
			<div className="login-content-holder content-section implementation">
				<div className="content-section implementation">
					<Title />
					<Card title={this.title} subTitle={this.subTitle} style={{ width: "360px" }} className="ui-card-shadow" footer={this.footer}>
						<div>
							<form
								onSubmit={e => {
									this.handleSubmit(true, e);
								}}
								action={"javascript:void(0);"}
							>
								<Input name="username" type="text" label="Username" required />
								<Input name="password" label="Password" type="password" required />
								<Button label="üîë  Login" type="submit" />
							</form>
						</div>
					</Card>
				</div>
			</div>
		);
	}
}

const Title = () => (
	<h1 style={{ fontSize: "50px" }}>
		üçÖ
		<span> Pomodro</span>
	</h1>
);
