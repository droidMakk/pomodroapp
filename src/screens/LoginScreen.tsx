import React, { Component } from 'react'
import { Input } from '../components/common';

export class LoginScreen extends Component {
  render() {
    return (
      <div>
        <Input label="Username" />
      </div>
    )
  }
}
