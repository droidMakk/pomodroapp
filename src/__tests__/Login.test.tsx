import React from 'react';
import renderer,{ ReactTestRenderer, ReactTestRendererTree } from 'react-test-renderer';
import { LoginScreen, ILoginScreen } from '../screens/Login/LoginScreen';
import { Card } from 'primereact/card';

test('Renders Login Screen Correctly',() => {
    
    const component:ReactTestRenderer = renderer.create(
        <LoginScreen />
    )
    
    let content = component.root.findByProps({ className: "content-section implementation" });
    let cardChildren = content.findAllByType(Card);
    let passwordField = cardChildren[0].findByProps({ label: "Password" })

    
    console.info('PASSWORD FIELD',passwordField);
    passwordField.props.onChange({ currentTarget: { value: 'droidmakk' } })
    
    expect(passwordField.props.value).toBe('droidmakk');
    
})