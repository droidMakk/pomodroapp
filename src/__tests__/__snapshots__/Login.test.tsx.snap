// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input Fields 1`] = `
<div
  className="login-content-holder content-section implementation"
>
  <div
    className="content-section implementation"
  >
    <div
      className="p-card p-component ui-card-shadow"
      style={
        Object {
          "width": "360px",
        }
      }
    >
      <div
        className="p-card-header"
      >
        header
      </div>
      <div
        className="p-card-body"
      >
        <div
          className="p-card-title"
        >
          Login
        </div>
        <div
          className="p-card-subtitle"
        >
          Enter your credentials to get started!
        </div>
        <div
          className="p-card-content"
        >
           
          <div>
            <span
              className="p-float-label"
            >
              <input
                className="p-inputtext p-component"
                id="float-input"
                label="Username"
                onChange={[Function]}
                onInput={[Function]}
                onKeyPress={[Function]}
              />
              <label
                htmlFor="float-input"
              >
                Username
              </label>
            </span>
            <span
              className="p-float-label"
            >
              <input
                className="p-inputtext p-component"
                id="float-input"
                label="Password"
                onChange={[Function]}
                onInput={[Function]}
                onKeyPress={[Function]}
                type="password"
              />
              <label
                htmlFor="float-input"
              >
                Password
              </label>
            </span>
          </div>
           
        </div>
        <div
          className="p-card-footer"
        >
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
